plugin/storage/badger/options.go

```diff
@@ -160,6 +164,9 @@ func initFromViper(cfg *NamespaceConfig, v *viper.Viper) {
 	cfg.MetricsUpdateInterval = v.GetDuration(cfg.namespace + suffixMetricsInterval)
 	cfg.Truncate = v.GetBool(cfg.namespace + suffixTruncate)
 	cfg.ReadOnly = v.GetBool(cfg.namespace + suffixReadOnly)
+	if cfg.Truncate {
```

Comment:
nit: using `v.IsSet(cfg.namespace + suffixTruncate)` would notify users of the deprecation if they were explicitly doing `--badger.truncate=false`, even though it's redundant, but who knows. :)